worker/worker.go

```diff
@@ -519,6 +519,54 @@ func (w *worker) rhpScanHandler(jc jape.Context) {
 	})
 }
 
+func (w *worker) fetchActiveContracts(ctx context.Context, metadatas []api.ContractMetadata, timeout time.Duration, bh uint64) (contracts []api.Contract, errs HostErrorSet) {
+	// fill requests channel
+	reqs := make(chan api.ContractMetadata, len(metadatas))
+	for _, metadata := range metadatas {
+		reqs <- metadata
+	}
+	close(reqs)
+
+	// prepare response channel
+	type res struct {
+		Contract api.Contract
+		Err      *HostError
+	}
+	resps := make(chan res, len(metadatas))
```

Comment:
I'm personally not a huge fan of the double buffered channel pattern here.
We really only need 1 unbuffered channel here to send requests to the workers.

If you don't like having workers lock a slice and append to it, you could preallocate a `resps := make([]res, len(metadata))` slice and pass an index in a `req` type. Then every worker would fetch the revision and finally write the response to the right index of the `resps` slice.

Reply:
Hm ok, I guess the buffered response channel is a bit odd but I personally don't really mind it. I don't mind locking a slice at all, I just noticed that you were appending to the `errs` outside the lock I think. I updated it to append under lock.